base/dates/parse.jl

```diff
@@ -1,7 +1,8 @@
 ### Parsing utilities
 
-@generated function tryparse_internal{T<:TimeType, S, N}(::Type{T}, str::AbstractString, df::DateFormat{S, NTuple{N}}, raise::Bool=false)
-    token_types = Type[dp <: DatePart ? SLOT_RULE[first(dp.parameters)] : Void for dp in df.parameters[2].parameters]
+@generated function tryparse_internal{T<:TimeType,S,F}(::Type{T}, str::AbstractString, df::DateFormat{S,F}, raise::Bool=false)
+    token_types = Type[dp <: DatePart ? SLOT_RULE[first(dp.parameters)] : Void for dp in F.parameters]
+    N = length(F.parameters)
```

Comment:
I preferred to keep minimal logic in the `@generated` function as it makes it easier to understand...